### PR TITLE
Remove usage of deprecated 'WebRequest'

### PR DESCRIPTION
**Pull request by Bond-009**
Ref: https://docs.microsoft.com/en-us/dotnet/api/system.net.webrequest?view=netframework-4.7.2

**Issues**
Maybe #465

### CODE REVIEW

**Comment by daniel-widrick** [comment]
I've got an hour or so before the weekend starts... 

Let me do a quick clone and test out the changes... See if this affects: https://github.com/jellyfin/jellyfin/issues/465

File: MediaBrowser.Common/Net/IHttpClient.cs [hunk 1]
```diff
@@ -23,13 +24,23 @@ public interface IHttpClient
         Task<Stream> Get(HttpRequestOptions options);
 
         /// <summary>
+        /// Warning: Depricated function,
```
Comment (cvium) [inline MediaBrowser.Common/Net/IHttpClient.cs]:
```suggestion
        /// Warning: Deprecated function,
```

File: Emby.Server.Implementations/HttpClientManager/HttpClientManager.cs [hunk 2]
```diff
@@ -74,146 +74,122 @@ public class HttpClientManager : IHttpClient
...
         {
-            if (string.IsNullOrEmpty(host))
-            {
-                throw new ArgumentNullException(nameof(host));
-            }
-
-            var key = host + enableHttpCompression;
+            var key = GetHostFromUrl(url) + enableHttpCompression;
 
             if (!_httpClients.TryGetValue(key, out var client))
```
Comment (cvium) [inline Emby.Server.Implementations/HttpClientManager/HttpClientManager.cs]: Shouldn't we be using the HttpClientFactory?

File: Emby.Server.Implementations/ApplicationHost.cs [hunk 3]
```diff
@@ -1479,12 +1479,12 @@ public async Task<string> GetWanApiUrl(CancellationToken cancellationToken)
                     LogErrorResponseBody = false,
                     LogErrors = false,
                     LogRequest = false,
-                    TimeoutMs = 10000,
```
Comment (JustAMan) [inline Emby.Server.Implementations/ApplicationHost.cs]: why remove?

File: Emby.Server.Implementations/ApplicationHost.cs [hunk 4]
```diff
@@ -1604,16 +1604,15 @@ private async Task<bool> IsIpAddressValidAsync(IpAddressInfo address, Cancellati
                     LogErrorResponseBody = false,
                     LogErrors = logPing,
                     LogRequest = logPing,
-                    TimeoutMs = 5000,
```
Comment (JustAMan) [inline Emby.Server.Implementations/ApplicationHost.cs]: again, why remove?

File: Emby.Server.Implementations/LiveTv/Listings/SchedulesDirect.cs [hunk 5]
```diff
@@ -96,8 +96,6 @@ public async Task<IEnumerable<ProgramInfo>> GetProgramsAsync(ListingsProviderInf
                 Url = ApiUrl + "/schedules",
                 UserAgent = UserAgent,
                 CancellationToken = cancellationToken,
-                // The data can be large so give it some extra time
-                TimeoutMs = 60000,
```
Comment (JustAMan) [inline Emby.Server.Implementations/LiveTv/Listings/SchedulesDirect.cs]: again, why remove?

File: Emby.Server.Implementations/LiveTv/Listings/SchedulesDirect.cs [hunk 6]
```diff
@@ -115,9 +113,7 @@ public async Task<IEnumerable<ProgramInfo>> GetProgramsAsync(ListingsProviderInf
                     Url = ApiUrl + "/programs",
                     UserAgent = UserAgent,
                     CancellationToken = cancellationToken,
-                    LogErrorResponseBody = true,
-                    // The data can be large so give it some extra time
-                    TimeoutMs = 60000
```
Comment (JustAMan) [inline Emby.Server.Implementations/LiveTv/Listings/SchedulesDirect.cs]: why?

File: Emby.Server.Implementations/LiveTv/Listings/SchedulesDirect.cs [hunk 7]
```diff
@@ -483,8 +479,6 @@ private static double GetAspectRatio(ScheduleDirect.ImageData i)
                 CancellationToken = cancellationToken,
                 RequestContent = imageIdString,
                 LogErrorResponseBody = true,
-                // The data can be large so give it some extra time
-                TimeoutMs = 60000
```
Comment (JustAMan) [inline Emby.Server.Implementations/LiveTv/Listings/SchedulesDirect.cs]: why?

File: Emby.Server.Implementations/LiveTv/Listings/SchedulesDirect.cs [hunk 8]
```diff
@@ -871,8 +865,6 @@ public async Task<List<ChannelInfo>> GetChannels(ListingsProviderInfo info, Canc
                 UserAgent = UserAgent,
                 CancellationToken = cancellationToken,
                 LogErrorResponseBody = true,
-                // The data can be large so give it some extra time
-                TimeoutMs = 60000
```
Comment (JustAMan) [inline Emby.Server.Implementations/LiveTv/Listings/SchedulesDirect.cs]: why?

File: Emby.Server.Implementations/LiveTv/TunerHosts/HdHomerun/HdHomerunHost.cs [hunk 9]
```diff
@@ -138,7 +138,6 @@ private async Task<DiscoverResponse> GetModelInfo(TunerHostInfo info, bool throw
                 {
                     Url = string.Format("{0}/discover.json", GetApiUrl(info)),
                     CancellationToken = cancellationToken,
-                    TimeoutMs = Convert.ToInt32(TimeSpan.FromSeconds(10).TotalMilliseconds),
```
Comment (JustAMan) [inline Emby.Server.Implementations/LiveTv/TunerHosts/HdHomerun/HdHomerunHost.cs]: why?

File: Emby.Server.Implementations/LiveTv/TunerHosts/HdHomerun/HdHomerunHost.cs [hunk 10]
```diff
@@ -191,7 +190,6 @@ private async Task<List<LiveTvTunerInfo>> GetTunerInfosHttp(TunerHostInfo info,
             {
                 Url = string.Format("{0}/tuners.html", GetApiUrl(info)),
                 CancellationToken = cancellationToken,
-                TimeoutMs = Convert.ToInt32(TimeSpan.FromSeconds(5).TotalMilliseconds),
```
Comment (JustAMan) [inline Emby.Server.Implementations/LiveTv/TunerHosts/HdHomerun/HdHomerunHost.cs]: why?

File: Emby.Server.Implementations/LiveTv/TunerHosts/SharedHttpStream.cs [hunk 11]
```diff
@@ -47,13 +47,7 @@ public override async Task Open(CancellationToken openCancellationToken)
                 CancellationToken = CancellationToken.None,
                 BufferContent = false,
 
-                // Increase a little bit
-                TimeoutMs = 30000,
```
Comment (JustAMan) [inline Emby.Server.Implementations/LiveTv/TunerHosts/SharedHttpStream.cs]: why?

File: Emby.Server.Implementations/LiveTv/TunerHosts/SharedHttpStream.cs [hunk 12]
```diff
@@ -47,13 +47,7 @@ public override async Task Open(CancellationToken openCancellationToken)
...
-
                 EnableHttpCompression = false,
-
-                LogResponse = true,
-                LogResponseHeaders = true
```
Comment (JustAMan) [inline Emby.Server.Implementations/LiveTv/TunerHosts/SharedHttpStream.cs]: why?

File: Emby.Server.Implementations/ApplicationHost.cs (hunk 3 above)
Comment (Bond-009) [reply Emby.Server.Implementations/ApplicationHost.cs]: Can't set the timeout for each request, only per httpclient.

File: Emby.Server.Implementations/HttpClientManager/HttpClientManager.cs (hunk 2 above)
Comment (Bond-009) [reply Emby.Server.Implementations/HttpClientManager/HttpClientManager.cs]: Maybe in the future? 🤷‍♂️

File: Emby.Server.Implementations/LiveTv/Listings/SchedulesDirect.cs (hunk 6 above)
Comment (JustAMan) [reply Emby.Server.Implementations/LiveTv/Listings/SchedulesDirect.cs]: I see why drop timeout (like "not supported"), but why drop logging error response?

Comment (Bond-009) [reply Emby.Server.Implementations/LiveTv/Listings/SchedulesDirect.cs]: The property was being set, but is wasn't actually being used iirc.

File: Emby.Server.Implementations/LiveTv/TunerHosts/SharedHttpStream.cs (hunk 12 above)
Comment (Bond-009) [reply Emby.Server.Implementations/LiveTv/TunerHosts/SharedHttpStream.cs]: Same, It wasn't being used.

**Comment by joshuaboniface** [comment]
Some build failures:

```
ApplicationHost.cs(1646,37): error CS0103: The name 'logPing' does not exist in the current context [/jellyfin/Emby.Server.Implementations/Emby.Server.Implementations.csproj]
ApplicationHost.cs(1647,38): error CS0103: The name 'logPing' does not exist in the current context [/jellyfin/Emby.Server.Implementations/Emby.Server.Implementations.csproj]
ApplicationHost.cs(1636,24): error CS0219: The variable 'LogPing' is assigned but its value is never used [/jellyfin/Emby.Server.Implementations/Emby.Server.Implementations.csproj]
```

**Comment by anthonylavado** [comment]
Are those errors reported by @joshuaboniface relevant?

File: Emby.Server.Implementations/ApplicationHost.cs [hunk 13]
```diff
@@ -1642,17 +1643,17 @@ private async Task<bool> IsIpAddressValidAsync(IpAddressInfo address, Cancellati
                     {
                         Url = apiUrl,
                         LogErrorResponseBody = false,
-                        LogErrors = LogPing,
-                        LogRequest = LogPing,
-                        TimeoutMs = 5000,
+                        LogErrors = logPing,
```
Comment (DrPandemic) [inline Emby.Server.Implementations/ApplicationHost.cs]:
```suggestion
                        LogErrors = LogPing,
```

File: Emby.Server.Implementations/ApplicationHost.cs [hunk 14]
```diff
@@ -1642,17 +1643,17 @@ private async Task<bool> IsIpAddressValidAsync(IpAddressInfo address, Cancellati
...
+                        LogRequest = logPing,
```
Comment (DrPandemic) [inline Emby.Server.Implementations/ApplicationHost.cs]:
```suggestion
                        LogRequest = LogPing,
```

**Pull request merged**